### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by dlevenstein** [buzsakilab/buzcode#200]
We should have a tutorial script that uses the sample structures and shows the user how to load LFP, load spikes, use the basic I/O functions, and run a basic analysis.

**Comment by brendonw1** [comment]
This could be a good thing for someone to do in a new lab as they get to
know the system.  If no one else does this before hand, then I can have
people in my lab do this as they learn over the next few months.  Obviously
if someone does this first that's great.

**Comment by dlevenstein** [comment]
Another related thing that needs to be done is a function description list. Can be updated/improved as we add more functions to the repo.   

Just putting this here for later notice to self or others.

**Comment by brendonw1** [comment]
What approximate format should this take?  Maybe example problems experiments where people are walked through the relevant code?

**Comment by DavidTingley** [comment]
uploaded to dev branch.  @raswanson can upload her tutorial, I'll work on one for ratemapping

**Comment by dlevenstein** [comment]
We just discussed this a bit. Sounds like Rachel actually made a quick tutorial for Omid and a visiting postdoc a few months ago. We're going to make a "tutorials" folder and put it in there and then we can expand it.

The thinking at this point was having a few tutorials - one basic that includes things like how to load lfp, spikes, sessionInfo. And then a few specialized ones with things like lfp-spike analyses or detection or detecting place fields.

**Comment by DavidTingley** [comment]
We should also update the Wiki to reflect this folder and its contents

**Comment by dlevenstein** [comment]
Jinx

**Comment by DavidTingley** [comment]
@brendonw1 @evilrobotxoxo Do you guys have any students that might make for good alpha testers for tutorial scripts that we put together?

**Comment by brendonw1** [comment]
Yeah I think so... we may try some place cell/field analysis.  I assume that's a good direction to go for a pretty complete and yet basic analysis type?

**Comment by brendonw1** [comment]
Hey thanks David, this looks great.
Can I ask how you get the proper .behavior.mat for place?
Do you use this?:
https://github.com/buzsakilab/buzcode/blob/a12ded5f4a6bd538bbdb926ad7b901d4c13722c8/preprocessing/positionTracking/LEDTracking/bz_processConvertLED2Behav.m

**Comment by DavidTingley** [comment]
or this for optitrack systems, they should both produce that same *.behavior.mat file format (with additional Z and orientation data for optitrack)

https://github.com/buzsakilab/buzcode/blob/a12ded5f4a6bd538bbdb926ad7b901d4c13722c8/preprocessing/positionTracking/optitrack/bz_processConvertOptitrack2Behav.m

**Comment by evilrobotxoxo** [comment]
So everything here gets posted to Slack, but it doesn't work in the opposite direction, I guess. Anyway, yes, I will have a couple of students/technicians in the next few weeks who will need buzcode tutorials. In fact, I would appreciate a buzcode tutorial for myself, in terms of the standardized structs and etc.

**Comment by brendonw1** [comment]
Sadly I haven't spent the $13K for optitrack yet :)
But good to know which code to use

Thanks David

**Comment by DavidTingley** [comment]
@evilrobotxoxo I wouldn't mind using the slack channel more often but it hasn't really been adopted by many lab members here.  @dlevenstein we could make another push for using it?

**Comment by brendonw1** [comment]
Prob w slack is it’s yet another place to check. I prefer one place - which
for the moment predominantly email anyway

**Comment by evilrobotxoxo** [comment]
I don't have any strong preference for using slack over email or the github
web page. I had just noticed for the first time that messages on slack
don't get cross-posted to github, but the ones on github get cross-posted
to slack.

Lucas L. Sjulson, MD, PhD
Assistant Professor
Department of Psychiatry and Behavioral Sciences
Dominick P. Purpura Department of Neuroscience
Albert Einstein College of Medicine
Rose F. Kennedy Center, Room 227
1410 Pelham Parkway South, Bronx, NY 10461
cell: 646-528-9672
http://sjulsonlab.org

**Comment by brendonw1** [comment]
We're still a few weeks away from this... we're working on training on
spike sorting and I think we have motion tracking working well, but not yet
tried on a full-length video.

We'll loop back to this

**Comment by AntonioFR8** [comment]
I have started writing some documentation in the form of Contents scripts for each main folder. I will be uploading them this week. 
If anybody have strong opinion against I will also reorganize a bit the folders. Like creating a SharpWaveRipple folder inside LFP analysis were to put together related staff that is disperse now (i.e. in detectors)

**Comment by evilrobotxoxo** [comment]
I had my rotation student Andrea (not Andrea Cumpelik) try the buzcode
tutorials, and she was (understandably) unable to figure out that she was
supposed to run "compilefma" and "mex CCGHeart" first. Is there a reason we
don't have compiled mex files in buzcode?

Luke

Lucas L. Sjulson, MD, PhD
Assistant Professor
Department of Psychiatry and Behavioral Sciences
Dominick P. Purpura Department of Neuroscience
Albert Einstein College of Medicine
Rose F. Kennedy Center, Room 227
1410 Pelham Parkway South, Bronx, NY 10461
cell: 646-528-9672
http://sjulsonlab.org

**Comment by dlevenstein** [comment]
There's a function 'compileBuzcode' in the top folder. I thought we'd put in the readme that the user should run this before doing anything....... apparently we did not! I'll add it.

**Comment by dlevenstein** [comment]
(also, to my knowledge these files have to be compiled on the users computer)

**Comment by dlevenstein** [comment]
I just made some updates to the [readme](https://github.com/buzsakilab/buzcode/blob/master/readme.txt), that indicate the need to compileBuzcode first.  Feel free to make it more helpful as you see fit!

**Comment by evilrobotxoxo** [comment]
Thanks, Dan!

Lucas L. Sjulson, MD, PhD
Assistant Professor
Department of Psychiatry and Behavioral Sciences
Dominick P. Purpura Department of Neuroscience
Albert Einstein College of Medicine
Rose F. Kennedy Center, Room 227
1410 Pelham Parkway South, Bronx, NY 10461
cell: 646-528-9672
http://sjulsonlab.org

**Comment by DavidTingley** [comment]
As you have people go through it @brendonw1 @evilrobotxoxo let us know what other tutorials would be most useful to upload first.

**Comment by brendonw1** [comment]
Yes
... having issues earlier than all that now... but soon... I hope

**Comment by dlevenstein** [comment]
Add Thomas to this
